client/components/LearnerViewDirective.js

```diff
@@ -733,6 +739,33 @@ tie.directive('learnerView', [function() {
         };
 
         /**
+          * Highlights syntax errors in the coding UI
+          *
+          * @param {number} lineNumber
+          */
+        var highlightLine = function(lineNumber) {
+          var actualLineNumber = lineNumber - 1;
+          var codeLines = document.querySelectorAll('.CodeMirror-line');
+          if ($scope.isInDarkMode === true) {
+            codeLines[actualLineNumber].className +=
+              ' tie-syntax-error-line night-mode';
+          } else {
+            codeLines[actualLineNumber].className += ' tie-syntax-error-line';
```

Comment:
I think, instead of manipulating strings, doing something like classList.add() would be better -- since that's what you're trying to do conceptually. See https://developer.mozilla.org/en-US/docs/Web/API/Element/classList